### PR TITLE
Fix #build, was not compiling

### PR DESCRIPTION
**Pull request by hanneskaeufler**
self.build was not expecting a block, but needs to

### CODE REVIEW

File: src/sitemapper.cr [hunk 1]
```diff
@@ -22,7 +22,9 @@ module Sitemapper
   end
 
   def self.build
-    build(host: config.host, mex_urls: config.max_urls, use_index: config.use_index)
+    builder = Sitemapper::Builder.new(config.host, config.max_urls, config.use_index)
```
Comment (jwoertink) [inline src/sitemapper.cr:25]: The other build method already takes the block and sets it up. This should probably just be something like:

```crystal
def self.build(&block)
  build(host: config.host, mex_urls: config.max_urls, use_index: config.use_index, &block)
end
```

It's really just a convenience method for when someone doesn't want to pass options, and just use the configured defaults.

**Review by jwoertink — CHANGES_REQUESTED** [review]

**Comment by jwoertink** [comment]
Also, thanks for all the PRs! Glad to have someone to help out 😄

File: src/sitemapper.cr (hunk 1 above)
Comment (hanneskaeufler) [reply src/sitemapper.cr:25]: Yeah that's what I thought, but this doesn't compile. Looks like the `with builder` isn't forwarded correctly. Please note I'm on crystal `0.25.1`, though I hope that doesn't make a difference.

```crystal
  def self.build(&block)
    build(host: config.host, max_urls: config.max_urls, use_index: config.use_index, &block)
  end

  def self.build(host : String, max_urls : Int32, use_index : Bool)
    builder = Sitemapper::Builder.new(host, max_urls, use_index)
    with builder yield
    builder.generate
  end
```

```
Error in line 4: while requiring "./spec/sitemapper_spec.cr"

in spec/sitemapper_spec.cr:6: undefined method 'add'

      sitemaps = Sitemapper.build { add("/") }
```

**Comment by hanneskaeufler** [comment]
No biggy, pleasure to have found the shard :) I did integrate it (my fork of it) into my blog already based on your post. I had one more change to make to get it to compile in my codebase (https://github.com/hanneskaeufler/sitemapper/commit/e645a68132f7391c47c960e40c603432914437e5), haven't figured out why though, thats why I'm running a fork. I was hesitant to open an issue since I am unsure of the origin.

File: src/sitemapper.cr (hunk 1 above)
Comment (jwoertink) [reply src/sitemapper.cr:25]: hmmm... yeah, I just tried messing around with it, and I'm not getting much either. I found [this issue](https://github.com/crystal-lang/crystal/issues/6080), so I think the whole `with x yield` thing might either have issues, or they may end up getting rid of it... Until that happens, I'm thinking let's just roll with the duplication for now. Can you just add a note above the code that mentions why it's duplicated or whatever? Just so when I see it in a few weeks I'm not questioning lol.

Thanks!

Comment (hanneskaeufler) [reply src/sitemapper.cr:25]: Note added ✅

**Review by jwoertink — APPROVED** [review]

**Comment by jwoertink** [comment]
Thanks @hanneskaeufler ! I've merged and released v0.3.0

**Pull request merged**